recipes/cuda-nvcc/meta.yaml

```diff
@@ -0,0 +1,129 @@
+{% set name = "cuda-nvcc" %}
+{% set version = "12.0.76" %}
+{% set cuda_version = "12.0" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set target_name = "x86_64-linux" %}  # [linux64]
+{% set target_name = "ppc64le-linux" %}  # [ppc64le]
+{% set target_name = "sbsa-linux" %}  # [aarch64]
+{% set target_name = "x64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_nvcc/{{ platform }}/cuda_nvcc-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: 160ca9f8828f53daa4b2066c5361aacb45fa2575885f70c223cda1d11df53d6f  # [linux64]
+  sha256: dc5bd7eac57b32a9645c0c23c6a1ba8a9533e846eeec31414e1aa4230afee2f8  # [ppc64le]
+  sha256: 85c8c643ea2cfa398e1b5ffcd597f5bb3b738526c9e3fb8e39dd909c55345f36  # [aarch64]
+  sha256: f67406f44263f0e4a110a8cf0200bff1d3d0e32917db8a7c9e417a87eb7ca372  # [win]
+  patches:
+    nvcc.profile.patch  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: cuda-nvcc
+    build:
+      run_exports:
+        strong:
+          - __glibc >=2.17        # [linux]
+    files:
+      - bin      # [linux]
+      - include  # [linux]
+      - lib      # [linux]
+      - nvvm     # [linux]
+      - Library\nvvm  # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+      host:
+        - sysroot_{{ target_platform }} 2.17   # [linux]
+        - __glibc >=2.17                       # [linux]
+        - cuda-version {{ cuda_version }}
+      run:
+        - sysroot_{{ target_platform }} 2.17   # [linux]
+        - __glibc >=2.17                       # [linux]
+        - cuda-cudart-dev
+        - cuda-nvcc_{{ target_platform }} =={{ version }}
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+      run_constrained:
+        - gcc_impl_{{ target_platform }} >=6,<13  # [linux]
```

Comment:
Since we only use `clang` on macOS and that is skipped, think we don't need to worry about that.

On Windows we don't distribute compiler packages and instead rely on CI to have them. Not sure what more we can do there

Reply:
> Since we only use `clang` on macOS and that is skipped, think we don't need to worry about that.

A user could presumably install it and want to use it in their environment though?

Reply:
Sorry am not following. How would they install it? This recipe isn't built on macOS

Reply:
There's linux-64 clang builds that someone could install and use in their environment.

Reply:
Ah ok. That makes more sense. Thanks Keith! 🙏

There is a `clang` package. Though the package that encodes `target_platform` would be [in `clang-compiler-activation-feedstock`]( https://github.com/conda-forge/clang-compiler-activation-feedstock/blob/d79fc38aef6f7f267a02623de29cd83f42830bdc/recipe/meta.yaml#L36 ). However this is only configured for macOS targets currently.

That said, suppose one could add Linux targets for `clang`. Maybe we could save this as a follow up issue (for whenever those packages get added)?

Reply:
We can always repodata patch at that time. I'm fine to defer it to a follow up.